lib/Dialect/FIRRTL/Transforms/CreateSiFiveMetadata.cpp

```diff
@@ -263,8 +460,24 @@ void CreateSiFiveMetadataPass::getDependentDialects(
 }
 
 void CreateSiFiveMetadataPass::runOnOperation() {
+  auto circuitOp = getOperation();
+  auto *body = circuitOp.getBody();
+
+  // Find the device under test and create a set of all modules underneath it.
+  auto it = llvm::find_if(*body, [&](Operation &op) -> bool {
+    return AnnotationSet(&op).hasAnnotation(dutAnnoClass);
+  });
+  if (it != body->end()) {
+    dutMod = dyn_cast<FModuleOp>(*it);
+    auto instanceGraph = getAnalysis<InstanceGraph>();
+    auto *node = instanceGraph.lookup(&(*it));
+    llvm::for_each(llvm::depth_first(node), [&](InstanceGraphNode *node) {
+      dutModuleSet.insert(node->getModule());
+    });
+  }
+  auto memFail = emitMemoryMetadata();
   if (failed(emitRetimeModulesMetadata()) ||
-      failed(emitSitestBlackboxMetadata()))
+      failed(emitSitestBlackboxMetadata()) || failed(memFail))
```

Comment:
```c++
  if (failed(emitRetimeModulesMetadata()) ||
      failed(emitSitestBlackboxMetadata()) ||
      failed(emitMemoryMetadata()))
```

Reply:
If `emitRetimeModulesMetadata` metadata fails, `emitMemoryMetadata` wouldn't run right ? Is that an expected behavior ? Do we want to emit rest of the metadata even if one of them fails.

Reply:
When we call `signalPassFailure()` we stop running pases and firtool won't print anything, so there isn't a point in printing the rest of the metadata

Reply:
Okay, that makes sense.